src/firewall.c

```diff
@@ -165,7 +179,18 @@ fw_refresh_client_list(void)
 			now = time(NULL);
 			last_updated = cp1->counters.last_updated;
 			added_time = cp1->added_time;
-			if (last_updated +  (config->checkinterval * config->clienttimeout) <= now) {
+
+			if (is_arp_valid() && !arp_get(ip)) {
```

Comment:
arp_get() allocates memory that must be freed by the caller. This is a memory leak. :P